transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java

```diff
@@ -361,17 +472,57 @@ public void onError(@NonNull Errors errors) {
     }
 
     @VisibleForTesting
-    void loadTransferDestination(@NonNull final String sourceToken) {
+    void loadPrepaidCardList(@NonNull final ArrayList<TransferSourceWrapper> sources) {
         mIsLoading.postValue(Boolean.TRUE);
-        mTransferMethodRepository.loadLatestTransferMethod(new TransferMethodRepository.LoadTransferMethodCallback() {
+        mPrepaidCardRepository.loadPrepaidCards(new PrepaidCardRepository.LoadPrepaidCardsCallback() {
             @Override
-            public void onTransferMethodLoaded(@Nullable TransferMethod transferMethod) {
-                mTransferDestination.postValue(transferMethod);
-                if (transferMethod == null) { // dismiss quote
-                    mIsLoading.postValue(Boolean.FALSE);
-                } else {
-                    quoteAvailableTransferFunds(sourceToken, transferMethod);
+            public void onPrepaidCardListLoaded(@Nullable List<PrepaidCard> prepaidCardList) {
+                mIsLoading.postValue(Boolean.FALSE);
+                if (prepaidCardList != null) {
+                    if (prepaidCardList.size() > 1) {
+                        sortPrepaidCard(prepaidCardList);
+                    }
+                    for (PrepaidCard prepaidCard : prepaidCardList) {
+                        TransferSourceWrapper sourceWrapper = new TransferSourceWrapper();
+                        sourceWrapper.setToken(prepaidCard.getField(TOKEN));
+                        sourceWrapper.setType(PREPAID_CARD);
+                        sourceWrapper.setIdentification(prepaidCard);
+                        sources.add(sourceWrapper);
+                    }
                 }
+                mSelectedTransferSource.postValue(sources.get(0));
+                mSourceToken = sources.get(0).getToken();
+                loadTransferDestination(mSourceToken);
+                mTransferSources.postValue(sources);
+            }
+
+            @Override
+            public void onError(@NonNull Errors errors) {
+                mIsLoading.postValue(Boolean.FALSE);
+                mLoadTransferRequiredDataErrors.postValue(new Event<>(errors));
+            }
+        });
+    }
+
+    public void loadTransferDestinationListWithoutPPC(@NonNull final String sourceToken) {
+        mIsLoading.postValue(Boolean.TRUE);
+        mTransferMethodRepository.loadTransferMethods(new TransferMethodRepository.LoadTransferMethodListCallback() {
+            @Override
+            public void onTransferMethodListLoaded(List<TransferMethod> transferMethods) {
+                if (transferMethods != null) {
+                    for (int i = transferMethods.size() - 1; i >= 0; --i) {
+                        if (Objects.equals(transferMethods.get(i).getField(TYPE), PREPAID_CARD)) {
+                            transferMethods.remove(i);
+                        }
+                    }
```

Comment:
this might throw `ConcurrentModificationException` use iterator

Reply:
fixed